docs/mvvm/MigratingFromMvvmLight.md

```diff
@@ -0,0 +1,842 @@
+
+Below are a list of migrations that will need to be performed if being used in your current solution. Where a method or property isn't listed, there is a direct replacement with the same name in the MVVM Toolkit and there is no change required.
+
+The first change here will be swapping using directives in your components.
+
+```csharp
+// MvvmLight
+using GalaSoft.MvvmLight.Command;
+using Galasoft.MvvmLight.CommandWpf;
+
+// Toolkit.Mvvm
+using Microsoft.Toolkit.Mvvm.Input;
+```
+
+**Note** on `RelayCommand<T>` constructors. MvvmLight uses weak references to establish the link between the command and the action called from the associated class. This is not required by the MVVM Toolkit implementation and if this optional parameter has been set to `true` in any of your constructors, this will be removed.
+
+### Using RelayCommand with asynchronous actions
+
+If you are currently using the MvvmLight `RelayCommand<T>` implementation with asynchronous actions, the MVVM Toolkit exposes an improved implementation for these scenarios. 
+
+You can simply replace your existing `RelayCommand<T>` with the `AsyncRelayCommand<T>` which has been built for asynchronous purposes.
+
+```csharp
+// MvvmLight
+var command = new RelayCommand<string>(async () => await this.OnCommandAsync());
+
+// Toolkit.Mvvm
+var asyncCommand = new AsyncRelayCommand<string>(this.OnCommandAsync);
+```
+
+### RelayCommand<T> Methods
+
+#### RaiseCanExecuteChanged ()
+
+The functionality of `RaiseCanExecuteChanged()` can be achieved with the MVVM Toolkit's `NotifyCanExecuteChanged()` method. 
+
+```csharp
+// MvvmLight
+var command = new RelayCommand<string>(this.OnCommand);
+command.RaiseCanExecuteChanged();
+
+// Toolkit.Mvvm
+var command = new RelayCommand<string>(this.OnCommand);
+command.NotifyCanExecuteChanged();
+```
+
+## Migrating SimpleIoc
+
+The [IoC](Ioc) implementation in the MVVM Toolkit takes advantage of existing .NET APIs through the `Microsoft.Extensions.DependencyInjection` library. 
+
+This is the biggest change between MvvmLight and the MVVM Toolkit. 
+
+This implementation will feel familiar if you've implemented dependency injection with ASP.NET Core applications.
+
+### Registering your dependencies
+
+With MvvmLight, you may have registered your dependencies similar to these scenarios using `SimpleIoc`.
+
+```csharp
+public void RegisterServices()
+{
+  SimpleIoc.Default.Register<INavigationService, NavigationService>();
+
+  SimpleIoc.Default.Register<IDialogService>(() => new DialogService());
+}
+```
+
+With the MVVM Toolkit, you would achieve the same as follows.
+
+```csharp
+public void RegisterServices()
+{
+  Ioc.Default.ConfigureServices(services => 
+  {
+    services.AddSingleton<INavigationService, NavigationService>();
+
+    services.AddSingleton<IDialogService>(new DialogService());
+  });
```

Comment:
We need to account for us not having the `Microsoft.Extensions.DependencyInjection` dependency anymore now. We should mention developers can install whatever DI library they prefer, such as that one, and then they can use that to initialize the `Ioc` class. Here's an example using `MS.Ex.DI`:

```suggestion
 Ioc.Default.ConfigureServices(
    new ServiceCollection()
    .AddSingleton<INavigationService, NavigationService>()
    .AddSingleton<IDialogService>(new DialogService())
    .BuildServiceProvider());
```